addons/web_editor/controllers/main.py

```diff
@@ -548,29 +548,46 @@ def subber(match):
             ('Cache-control', 'max-age=%s' % http.STATIC_CACHE_LONG),
         ])
 
-    @http.route(['/web_editor/media_library_url'], type='json', auth="user", website=True)
-    def media_library_url(self):
-        dbuuid = request.env['ir.config_parameter'].sudo().get_param('database.uuid')
+    @http.route(['/web_editor/media_library_search'], type='json', auth="user", website=True)
+    def media_library_search(self, **params):
         endpoint = request.env['ir.config_parameter'].sudo().get_param('web_editor.media_library_endpoint', DEFAULT_LIBRARY_ENDPOINT)
-        return '%s?dbuuid=%s' % (endpoint, dbuuid)
+        params['dbuuid'] = request.env['ir.config_parameter'].sudo().get_param('database.uuid')
+        response = requests.get('%s/1/search' % endpoint, params=url_encode(params))
+        if response.status_code == requests.codes.ok:
+            return response.json()
+        else:
+            return {'error': response.status_code}
 
-    @http.route('/web_editor/save_media', type='json', auth='user', methods=['POST'])
-    def save_media(self, media=[]):
+    @http.route('/web_editor/save_library_media', type='json', auth='user', methods=['POST'])
+    def save_library_media(self, media=[]):
         """
         Saves images from the media library as new attachments, making them
         dynamic SVGs if needed.
-            media = [
-                {
-                    'media_url': 'https://...',
+            media = {
+                <media_id>: {
                     'query': 'space separated search terms',
                     'is_dynamic_svg': True/False,
                 }, ...
-            ]
+            }
         """
         attachments = []
+        library_endpoint = request.env['ir.config_parameter'].sudo().get_param('web_editor.media_library_endpoint', DEFAULT_LIBRARY_ENDPOINT)
+
+        media_ids = ','.join([key for key, value in media])
+        params = {
+            'dbuuid': request.env['ir.config_parameter'].sudo().get_param('database.uuid'),
+            'media_ids': media_ids,
+        }
+        response = requests.get('%s/1/download_urls' % library_endpoint, params=url_encode(params))
+        if response.status_code != requests.codes.ok:
+            logger.exception("ERROR: couldn't get download urls from media library.")
+            raise Exception(_("ERROR: couldn't get download urls from media library."))
```

Comment:
raise exception, will pobably already log something gin the log.
And avoid duplicate string, you can just use var if my first supposition is not true

Reply:
I took the same approach as in web_unsplash, I thought the idea was that the first line wouldn't be translated, but since the line raising the exception is logged in the traceback, there isn't much of a point. Removed the duplication :)